src/mono/sample/HelloWorld/HelloWorld.csproj

```diff
@@ -16,6 +16,7 @@
 
     <ItemGroup>
       <AotInputAssemblies Include="$(PublishDir)\*.dll" />
+      <ReferenceAssembliesForPGO Include="$(PublishDir)\*.dll" />
```

Comment:
Do you need full paths to files or directories?

Reply:
I think both work. I originally passed in the whole directory, but it would lead to even `.aotdata` files being passed in. As a result, this would lead to a bunch of warnings and even one error with `System.Private.CoreLib.aotdata` not being loaded properly. https://gist.github.com/mdh1418/f73993031669311e23fb49e2f39c61ce
Because we are running the dotnet-pgo executable as a process, the warnings and error end up with an exit code 244. (Which is odd because running the command/args manually leads to exit code 0 😕)

To keep things simple, I just specified `.dll` since that is what I think we only care about when using dotnet-pgo on a .nettrace. However, we would need to modify this if we want to load non `.dll` as well (which im not sure if thats even relevant).

Unfortunately, I think this leads to a great number of undesired assemblies being referenced. I'm not sure if I understand correctly, but I think that for the samples, only `System.Private.CoreLib.dll`, `System.Runtime.dll`, `<Sample>.dll` are really the only assemblies needed to be referenced for the tool (So whatever assemblies were needed for the program/app to run). I'm not sure if theres a huge impact on performance by supplying ALL of the `.dll` in the publish file, and maybe we can pursue getting the exact assemblies as an optimization step.

Reply:
> .. [gist.github.com/mdh1418/f73993031669311e23fb49e2f39c61ce](https://gist.github.com/mdh1418/f73993031669311e23fb49e2f39c61ce) Because we are running the dotnet-pgo executable as a process, the warnings and error end up with an exit code 244. (Which is odd because running the command/args manually leads to exit code 0 😕)

I can improve this so the errors, and warnings from dotnet-pgo get logged as errors/warnings. I'll open a PR tomorrow.

Reply:
From what I found, both adding a list of references like you are doing and sending in a wildcard string will work.

For example, if you have an item that has the value `<path>\*.dll`, then the tool will enumerate all files. That might be better in this case.

https://github.com/dotnet/runtime/blob/e70b6dc04ea4d304dee91312bc5efc17d9d3cd41/src/coreclr/tools/Common/CommandLine/CommandLineHelpers.cs#L22-L30

Reply:
I think you still keep it an item in the event you have other directories with assemblies you want to include.  Both what was merged and what I'm suggesting work.  At the end of the day, I don't have a really strong opinion.